tests/e2e/ctl_v3_auth_cluster_test.go

```diff
@@ -0,0 +1,138 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/e2e"
+	"sync"
+	"testing"
+)
+
+const (
+	numKeys      = 10
+	writeThreads = 10
+
+	rootUser     = "root"
+	rootPassword = "root"
+	rootRole     = "root"
+	testUser     = "test"
+	testPassword = "testPassword"
+	testRole     = "testRole"
+)
+
+func TestAuthCluster(t *testing.T) {
+	e2e.BeforeTest(t)
+	cfg := e2e.NewConfigNoTLS()
+	cfg.ClusterSize = 1
+	cfg.SnapshotCount = 2
+	cfg.UserName = rootUser
+	cfg.Password = rootPassword
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	epc, err := e2e.NewEtcdProcessCluster(ctx, t, cfg)
+	if err != nil {
+		t.Fatalf("could not start etcd process cluster (%v)", err)
+	}
+	defer func() {
+		if err := epc.Close(); err != nil {
+			t.Fatalf("could not close test cluster (%v)", err)
+		}
+	}()
+
+	epcClient := epc.Client()
+	createUsers(ctx, t, epcClient)
+
+	if _, err := epcClient.AuthEnable(ctx); err != nil {
+		t.Fatalf("could not enable Auth: (%v)", err)
+	}
+
+	writeKeys(ctx, epc.Client().WithAuth(testUser, testPassword), "test")
+
+	if err := epc.StartNewProc(ctx, t); err != nil {
+		t.Fatalf("could not start second etcd process (%v)", err)
+	}
+
+	writeKeys(ctx, epc.Client().WithAuth(testUser, testPassword), "test_two_nodes")
+
+	hashKvs, err := epc.Client().HashKV(ctx, numKeys*2)
+	if err != nil {
+		t.Fatalf("could not HashKV (%v)", err)
+	}
+
+	revisionSet := false
+	var revision int64
+	var hash uint32
+	for _, v := range hashKvs {
+		if !revisionSet {
+			revision = v.Header.GetRevision()
+			hash = v.Hash
+			revisionSet = true
+			continue
+		}
+		if revision != v.Header.GetRevision() || hash != v.Hash {
+			t.Errorf("Inconsistent revisions found:")
+			for _, v := range hashKvs {
+				t.Errorf("%+v", v)
+			}
+			t.Fail()
+		}
+	}
```

Comment:
Based on the info in https://github.com/etcd-io/etcd/issues/14571, when adding a new member, it fails to recover from a snapshot.  But I do not see you add any new member in this case.

Reply:
it recovers successfully with an empty rangePermCache which leads to auth failures if client connects to the new node or corrupted state if state is modified on leader.